configure-c8y.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/sh
+
+
+if [ `id -u` != 0 ] ; 
+  then printf "Please run as sudo or root!\n"
+  exit
+fi
+printf "\n*********************************************"
+printf "\nThis script lets you setup a thinEdge connection to cumulocity IoT.\nWe assume that the installation was successful and all the necessary services are running.
+Just follow the instructions....."
+printf "\n*********************************************"
+
+printf "\n\n"
+read  -p 'Enter the Cumulocity tenant url e.g. [your-tenant.cumulocity.com]: ' tenant_url
+tenant_url=${tenant_url:-your-tenant.cumulocity.com}
+tedge config set c8y.url $tenant_url
+
+printf "\n\n"
+read -p 'Enter the path to your root certificates. On most Linux systems that is [/etc/ssl/certs]: ' path_root_cert
+path_root_cert=${path_root_cert:-/etc/ssl/certs}
+tedge config set c8y.root.cert.path $path_root_cert
+
+printf "\n\n"
+read -p 'Enter your device-id [my-device]: ' device_id
+device_id=${device_id:-my-device}
+{
+    tedge cert create --device-id $device_id
+} ||
+printf "\n\n"
+read -r -p "Would you like to remove the old cert and create a new one? (y)Yes/(n)No: " response
+case $response in
+    [yY]) 
+        printf "\nRemoving cert...."
+        tedge cert remove
+        tedge cert create --device-id $device_id
+        ;;
+    *) 
+        printf "\nOk bye, enjoy thingEdge.io!!!\n" 
+        exit 
+        ;;    
+esac
+
+
+printf "\n\nYour current configuration:\n"
+tedge config list
+
+
+RET=1
+until [ ${RET} -eq 0 ]; do
+    printf "\n\n"
+    read -p "Enter Cumulocity user [your@email.com]: " c8y_Username
+    c8y_Username=${c8y_Username:-your@email.com}
+    tedge cert upload c8y --user $c8y_Username
+    RET=$?
+done
+
+
+printf '\nCreated Certificate:'
+sudo tedge cert show
+
+
+{
+    printf "\nConnecting...."
+    tedge connect c8y
+} ||
```

Comment:
This syntax seems unnecessary to me.

If you would like to ignore errors for a specific command then it is better to limit the scope of the errors to that of the single command.

```suggestion
printf "\nConnecting...."
tedge connect c8y || true
```

However, I would have though it would be better to not ignore errors in the connection (as that should abort the execution) as there is no point going further if the previous command failed (imho)

Reply:
Actually I try to only execute the reconnect if tedge connect fails (e.g. already connected). Maybe I miss some brackets here.
What I don't get is where this error is coming from....

Certificate uploaded successfully.

Created Certificate:Device certificate: /etc/tedge/device-certs/tedge-certificate.pem
Subject: CN=my-device, O=Thin Edge, OU=Test Device
Issuer: CN=my-device, O=Thin Edge, OU=Test Device
Valid from: Sun, 04 Dec 2022 14:37:25 +0000
Valid up to: Mon, 04 Dec 2023 14:37:25 +0000
Thumbprint: 9C1FA57E5B21A3E790A2313EFCEEA7F6ED9175DB

Connecting....The system config file '/etc/tedge/system.toml' doesn't exist. Use '/bin/systemctl' as a service manager.

Checking if systemd is available.

Checking if configuration for requested bridge already exists.

Validating the bridge certificates.

Creating the device in Cumulocity cloud.

ERROR: Io(Custom { kind: InvalidData, error: AlertReceived(CertificateUnknown) })
Error: failed to connect Cumulocity cloud.

Caused by:
    Connection check failed


This happens when I do a disconnect before I run the script. When I then execute the reconnect in the script it works....

Reply:
It's a bit unclear what sequence you are talking about.

* What are the exact steps (showing happy and negative paths)
* Is it repeatable?
* `CertificateUnknown` error is generally when the cloud (I am going to guess Cumulocity) does not trust the certificate. You may need to disable/enable the certificate in Cumulocity if the cert is already showing that it is trusted (maybe a server side bug regarding the propagation of the trusted certificates to all cores?)

Reply:
it might be a timing issue. Usually you have some time between the upload and the start/restart. The script might be faster and core has not finished initializing the cert.... i will try another sleep here

Reply:
Used a timeout and the issue is gone. Fyi I do the retry to cover wrong password entries here.